libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KotlinProperties.kt

```diff
@@ -96,6 +97,16 @@ internal class PropertiesProvider private constructor(private val project: Proje
     val usePreciseJavaTracking: Boolean?
         get() = booleanProperty("kotlin.incremental.usePreciseJavaTracking")
 
+    val useClasspathSnapshot: Boolean
+        get() {
+            // The feature should be controlled by a Gradle property.
+            // Currently, we also allow it to be controlled by a system property to make it easier to test the feature during development.
+            // TODO: Remove the system property later.
+            val gradleProperty = booleanProperty("kotlin.incremental.useClasspathSnapshot") ?: false
+            val systemProperty = System.getProperty("kotlin.incremental.useClasspathSnapshot")?.toBoolean() ?: false
```

Comment:
Direct system property read will lead to configuration cache issues

Reply:
Thanks for reviewing this PR!

I tried to use the proper API `project.providers.systemProperty("kotlin.incremental.useClasspathSnapshot")` earlier, but tests failed because that API only exists for Gradle 6.5+, but the current Gradle version to run tests is 6.1.1.

I guess we can update access to all system properties (and Gradle properties) to the new API once we update to Gradle 6.5+.

If you still have a concern with this PR, please let me know.

Reply:
Can we have Gradle version dependent behaviour ?

Reply:
Yes, we can do that. But other properties in this class and other places in KGP already have this issue, so we should do that for all of them, and that would be a separate effort/PR right?

Reply:
I don't see `System.getProperty` used anywhere else in the class.

Reply:
I thought system properties and Gradle properties were the same when it comes to configuration-cache compatibility?

Reply:
Yes, eventually. However, ATM there are checks for system properties, not for Gradle properties.

Reply:
Ah okay, I'll upload a patch for this then.

Reply:
You can consider using `getSystemProperty` util function: https://github.com/JetBrains/kotlin/blob/33ddeffcfd68535e497e287ff1b7fef67a5e9ab3/libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/utils/configurationCache.kt#L19
Basically I think it's better not introduce new issues if there's easy solution to avoid them even if existing code has them :)

Reply:
Nice, I'm using your util method now.